packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -162,11 +162,17 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
         : startTextSelectionPoint.point.dy + _kToolbarHeight + _kToolbarContentDistanceBelow;
     final Offset preciseMidpoint = Offset(position.dx, y);
 
+    // get the Overlay's render box
```

Comment:
That the toolbar is placed in an overlay seems like an implementation detail of the `TextSelectionOverlay` class. Do we really want to have implementations of TextSelectionControls depend on that? Maybe `buildToolbar` should take some kind of a constraints parameter instead specifying how big the tool bar can be?

Reply:
I can look at doing this.
And to answer your larger picture question of whether there's a reason to have the toolbar in the child overlay... I think it makes more sense that way, as the toolbar will not start floating around far outside the actual child overlay's bounds, which could look kind of weird. After trying both ways for a few days, I can say that having it in the child overlay as it already is, and with the fix being presented here, it feels much more natural.

Reply:
@goderbauer Actually, I'm not sure if I should really be modifying the `TextSelectionControls`'s interface, since I imagine there can be different implementations of it for Web/desktop, etc that I'm not aware of, nor do I have any way of testing it. I'd be happy to have someone else take over from here, if that works.

Reply:
@goderbauer Any suggestions on this? Are the material and cupertino text_selection.dart files the only ones that implement that API? If so, I can make that change too.